state/protocol/badger/mutator.go

```diff
@@ -743,6 +743,10 @@ func (m *FollowerState) Finalize(ctx context.Context, blockID flow.Identifier) e
 		return fmt.Errorf("could not persist finalization operations for block (%x): %w", blockID, err)
 	}
 
+	// update the header cache
+	m.State.cachedFinal.Store(&cachedHeader{blockID, header})
+	m.State.cachedSealed.Store(&cachedHeader{lastSeal.BlockID, sealed})
```

Comment:
we are updating with two atomic value here.

I wonder if makes sense to use just one, because last seal and last finalized are always updated together:

```
type FinalizedSnapshot struct {
	Snapshot
	finalizedHeader *flow.Header
	sealedHeader *flow.Header
}
```

Reply:
How about this, instead? https://github.com/onflow/flow-go/pull/4145/commits/24e33e59b2a0fbf62ad413be5ee16c98938d375c. Reflecting that really they needn't always be updated together.

Reply:
yeah, that looks good too